src/approximate-qvm.lisp

```diff
@@ -0,0 +1,124 @@
+(in-package #:qvm)
+
+(defclass approx-qvm (channel-qvm)
+  ((t1-ops
+    :initarg :t1-ops
+    :accessor t1-ops
+    :initform (make-hash-table :test 'eql)
+    :documentation "t1 value to simulate")
+   (t2-ops
+    :initarg :t2-ops
+    :accessor t2-ops
+    :initform (make-hash-table :test 'eql)         
+    :documentation "t2 value to simulate")
+   (avg-gate-time
+    :initarg :avg-gate-time
+    :accessor avg-gate-time
+    :initform (error ":AVG-GATE-TIME is a required initarg to APPROX-QVM")
+    :documentation "The average gate time for a gate in this qvm")
+   (readout-povms
+    :initarg :readout-povms
+    :accessor readout-povms
+    :initform (make-hash-table :test 'eql))))
+
+
+(defun (setf qubit-t1) (t1 qvm qubit)
+  (let ((kraus-ops (generate-damping-kraus-ops t1 (avg-gate-time qvm))))
+    (check-kraus-ops kraus-ops)
+    (setf (gethash qubit (t1-ops qvm)) kraus-ops)))
+
+
+(defun (setf qubit-t2) (t2 qvm qubit)
+  "Save the kraus operators for the t2 provided on the specified qubit."
+  (let ((kraus-ops (generate-damping-dephasing-kraus-ops t2 (avg-gate-time qvm))))
+    (check-kraus-ops kraus-ops)
+    (setf (gethash qubit (t2-ops qvm)) kraus-ops))))
+
+
+(defmethod (setf qubit-fro) (fro qvm qubit)
+  "Save the readout povms for the fro provided on the specified qubit. The fR0 is assumed to be the average readout fidelity (avg of p(1|1) and p(0|0)) so as an approximation, we set p(0|0) = p(1|1) = fR0 in the readout povm assignment probabilities. "
+  (setf (gethash qubit (readout-povms qvm)) (list fro (- 1.0d0 fro) (- 1.0d0 fro) fro)))
+
+
+; For now just use a single gate time value for all calculations
```

Comment:
I will definitely add a more detailed comment here. Basically, the gate time is a parameter used in calculating the kraus operators for a T1 or T2 decay. Ideally, this gate-time value would be the gate time of the instruction preceding the application of the T1 or T2 kraus operators. Since I didn't know how to get the gate time for an qvm gate, for now, I decided to just let the gate time be a settable field in the approx-qvm.